### ISSUE TITLE
Wouldn't the source code for "Reset Connection Pool in case of destruction" be a problem?

### ISSUE THREAD

**Issue by orchsik** [pinojs/pino-elasticsearch#197]
### transform.destroy([error])

- Destroy the stream, and optionally emit an 'error' event. After this call, the transform stream would release any internal resources.
- Implementors should not override this method, but instead implement readable._destroy().
- The default implementation of _destroy() for Transform also emit 'close' unless emitClose is set in false.
- Once destroy() has been called, any further calls will be a no-op and no further errors except from _destroy() may be emitted as 'error'.

source = https://nodejs.org/api/stream.html#transformdestroyerror

---

"Implementors should not override this method, "

lib.js
```typescript
  // Resurrect connection pool on destroy
  splitter.destroy = () => {
    if (typeof client.connectionPool.resurrect === 'function') {
      client.connectionPool.resurrect({ name: 'elasticsearch-js' })
    }
  }
```
Wouldn't the source code for "Reset Connection Pool in case of destruction" be a problem?

**Comment by mahmoodhamdi** [comment]
Good catch — you're right that directly overriding `destroy` instead of implementing `_destroy` goes against the Node.js stream contract. The recommended approach is to use `_destroy` which is called internally by `destroy()`:

```js
splitter._destroy = (err, cb) => {
  if (typeof client.connectionPool.resurrect === "function") {
    client.connectionPool.resurrect({ name: "elasticsearch-js" })
  }
  cb(err)
}
```

The practical difference:
- **Overriding `destroy`**: Replaces the entire destroy logic — `close` event may not be emitted, internal cleanup is skipped, and the `emitClose` option is ignored
- **Using `_destroy`**: The stream's built-in `destroy()` handles the state machine (marking the stream as destroyed, emitting `close`, etc.) and calls `_destroy` for custom cleanup

In most cases the current override might work fine since the splitter stream is short-lived. But it's technically incorrect and could cause subtle issues like `close` events not firing, which would prevent proper cleanup in upstream consumers.

Would be a straightforward fix if you'd like to submit a PR for it.
